ocaml/xs-trace/xs_trace.ml

```diff
@@ -0,0 +1,35 @@
+let _ =
+  if Array.length Sys.argv <> 4 then (
+    Printf.eprintf "Usage: %s cp/mv <origin> <destination>\n" Sys.argv.(0) ;
+    exit 1
+  ) ;
+  let action = Sys.argv.(1) in
+  let origin = Sys.argv.(2) in
+  let url = Sys.argv.(3) in
```

Comment:
Parameter parsing is missing, currently it's not resistant to typos.
On top of that, allowing to _delete_ any directory in the system is really dangerous.
There are at least two options:
- Hardcode the root of the path that can be written to (and deleted)
- Use a list of allowed roots that can be used, the directories themselves cannot be erased
- 
```suggestion
  let action = match Sys.argv.(1) with | "mv" -> `mv | "cp" -> `cp | _ -> error_with_message usage in
  let origin =
    let param = Sys.argv.(2) in
    if ".." in param then invalid_argument "not allowed"
    Filename.concat root param
  in
```

The url can be parsed, but if it fails it worries me less as it will be very evident.